codice/ddf
# DDF-2491 Refactored Migrate and Replicate Commands to ingest initial query

#### What does this PR do?
The Migrate and Replicate commands would first query all the records to determine the total number of records. This PR refactors the Replicate and Migrate commands to issue one less query by ingesting the records returned from the first query.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@peterhuffer @glenhein @mweser 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire
@millerw8 
#### How should this be tested?
1. Confirm that TestReplicationCommand succeeds.
2. Confirm that Duplicate and Migrate console commands behave as expected.
 - Test different options (e.g. --batchsize, --multithreaded).
 - Confirm that the progress bar appears as expected.
 - Confirm that the source is not queried an extra time.

#### Any background context you want to provide?
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2491
#### Screenshots (if appropriate)
#### Checklist:
